src/main/java/io/github/jbellis/brokk/gui/Chrome.java

```diff
@@ -1025,13 +1026,34 @@ public void openFragmentPreview(ContextFragment fragment) {
             }
 
             // 6. Everything else (virtual fragments, skeletons, etc.)
-            var previewPanel = new PreviewTextPanel(contextManager,
-                                                    null,
-                                                    workingFragment.text(),
-                                                    workingFragment.syntaxStyle(),
-                                                    themeManager,
-                                                    workingFragment);
-            showPreviewFrame(contextManager, title, previewPanel);
+            if (workingFragment.isText() && workingFragment.syntaxStyle().equals(SyntaxConstants.SYNTAX_STYLE_MARKDOWN)) {
+                // Render as Markdown using a minimal TaskEntry wrapper
+                var markdownPanel = new MarkdownOutputPanel(false); // false = do not escape HTML
+                markdownPanel.updateTheme(themeManager.isDarkTheme());
+                
+                // Create a minimal TaskEntry to wrap the markdown content
+                var taskEntry = new TaskEntry(-1, 
+                    new ContextFragment.TaskFragment(contextManager, 
+                        List.of(dev.langchain4j.data.message.AiMessage.from(workingFragment.text())), 
+                        workingFragment.description()),
+                    null);
+                markdownPanel.setText(taskEntry);
```

Comment:
Maybe better to add `MarkdownOutputPanel.setText(String)` method and encapsulate the logic related to the helper/artificial `TaskFragment` in `MarkdownOutputPanel` (also change in other places where we used the same workaround)?